tests/foreman/cli/test_errata.py

```diff
@@ -80,113 +74,107 @@
+        }
+    )
 
-        :BZ: 1405428, 1418026, 1457977
-        """
-        super().setUpClass()
-        cls.org = make_org()
-        cls.env = make_lifecycle_environment({'organization-id': cls.org['id']})
-        cls.content_view = make_content_view({'organization-id': cls.org['id']})
-        cls.activation_key = make_activation_key(
-            {'lifecycle-environment-id': cls.env['id'], 'organization-id': cls.org['id']}
-        )
-        # add subscription to Satellite Tools repo to activation key
-        setup_org_for_a_rh_repo(
-            {
-                'product': PRDS['rhel'],
-                'repository-set': REPOSET['rhst7'],
-                'repository': REPOS['rhst7']['name'],
-                'organization-id': cls.org['id'],
-                'content-view-id': cls.content_view['id'],
-                'lifecycle-environment-id': cls.env['id'],
-                'activationkey-id': cls.activation_key['id'],
-            }
-        )
-        # create custom repository and add subscription to activation key
-        setup_org_for_a_custom_repo(
-            {
-                'url': cls.CUSTOM_REPO_URL,
-                'organization-id': cls.org['id'],
-                'content-view-id': cls.content_view['id'],
-                'lifecycle-environment-id': cls.env['id'],
-                'activationkey-id': cls.activation_key['id'],
-            }
-        )
 
-    def setUp(self):
-        """Create and setup host collection, hosts and virtual machines."""
-        super().setUp()
-        self.virtual_machines = []
-        self.host_collection = make_host_collection({'organization-id': self.org['id']})
-        for _ in range(self.VIRTUAL_MACHINES_COUNT):
-            # create VM
-            virtual_machine = VirtualMachine(distro=DISTRO_RHEL7)
-            virtual_machine.create()
-            self.addCleanup(vm_cleanup, virtual_machine)
-            self.virtual_machines.append(virtual_machine)
-            virtual_machine.install_katello_ca()
-            # register content host
-            virtual_machine.register_contenthost(
-                self.org['name'], activation_key=self.activation_key['name']
-            )
-            # enable red hat satellite repository
-            virtual_machine.enable_repo(REPOS['rhst7']['id'])
-            # install katello-agent
-            virtual_machine.install_katello_agent()
-            host = Host.info({'name': virtual_machine.hostname})
-            HostCollection.add_host(
-                {
-                    'id': self.host_collection['id'],
-                    'organization-id': self.org['id'],
-                    'host-ids': host['id'],
-                }
-            )
-        ActivationKey.add_host_collection(
+@pytest.fixture(scope='module')
+def setUp(module_org, module_activation_key):
+    """Create and setup host collection, hosts and virtual machines."""
+    virtual_machines = []
+    VIRTUAL_MACHINES_COUNT = 2
```

Comment:
I don't see any benefit to defining this attribute in this way.  I appreciate and like the idea of not just hard-coding a 'magic' number in the for loop, but it doesn't need to be capitalized like a constant.

To turn this into a 'feature' of the fixture, tie this vm count to `request.param`, so that tests can modify it with indirect parametrization.

Reply:
Hello

I have done that, please review once more
Thank you